### PR TITLE
ANDROID-11387 Support different types of icons in data card

### PR DESCRIPTION
**Pull request by jeslat**
### :goal_net: What's the goal?
Support different types of icons in the data card component.

You can see the spec here: https://www.figma.com/file/tKdPOfcUALzVIh5oizFbm7/%F0%9F%94%B8-Cards-Specs?node-id=201%3A7100&t=MI2ojg3d5fcxeDIS-0

### :construction: How do we do it?
* Update legacy implementation to support different types of icons
* Update compose implementation to support not circular icons in `IconPainter`
* Update both catalogs adding a dropdown to select the icon type

### ☑️ Checks
- [x] I updated the documentation (readmes, wikis, etc). If no need to update documentation explain why.
- [x] Tested with dark mode.
- [x] Tested with API 21.

### :test_tube: How can I test this?
- Catalog version: https://install.appcenter.ms/orgs/tuenti-organization/apps/mistica-catalog/distribution_groups/public
- Legacy implementation:
drawing
drawing
drawing
drawing

- Compose implementation:
drawing
drawing
drawing
drawing

### CODE REVIEW

File: catalog-compose/src/main/java/com/telefonica/mistica/compose/catalog/ui/components/DataCards.kt [hunk 1]
```diff
@@ -65,16 +59,16 @@ fun DataCards() {
     ) {
 
         OutlinedTextField(value = tag, onValueChange = { tag = it }, label = { Text("Tag") })
-        DropdownMenu(expanded = tagTypeDropDownExpanded, onDismissRequest = { tagTypeDropDownExpanded = false }) {
```
Comment (jeslat) [inline catalog-compose/src/main/java/com/telefonica/mistica/compose/catalog/ui/components/DataCards.kt:68]: This dropdown was not working, it was not visible at all. I've replaced it with the `DropDownInput`

File: catalog-compose/src/main/java/com/telefonica/mistica/compose/catalog/ui/components/DataCards.kt [hunk 2]
```diff
@@ -97,12 +91,30 @@ fun DataCards() {
             Checkbox(checked = withIcon, onCheckedChange = { withIcon = !withIcon })
         }
 
+        DropDownInput(
```
Comment (jeslat) [inline catalog-compose/src/main/java/com/telefonica/mistica/compose/catalog/ui/components/DataCards.kt:94]: New dropdown to select the icon type

File: library/src/main/java/com/telefonica/mistica/compose/README.md [hunk 3]
```diff
@@ -47,34 +49,26 @@ where `brand` is the implementation of the app brand and `darkTheme` a flag that
...
-* [Buttons](./src/main/java/com/telefonica/mistica/compose/button)
-* [Feedbacks](./src/main/java/com/telefonica/mistica/compose/feedback)
-* [Lists](./src/main/java/com/telefonica/mistica/compose/list)
-* [Tags](./src/main/java/com/telefonica/mistica/compose/tag)
-* [Carousel](./src/main/java/com/telefonica/mistica/compose/carousel)
-* [Empty States](./src/main/java/com/telefonica/mistica/compose/emptystate/screen)
-* [Empty State Cards](./src/main/java/com/telefonica/mistica/compose/emptystate/card)
-* [Inputs](./src/main/java/com/telefonica/mistica/compose/input)
-* [Steppers](./src/main/java/com/telefonica/mistica/compose/stepper)
-* [Title](./src/main/java/com/telefonica/mistica/compose/title)
```
Comment (jeslat) [inline library/src/main/java/com/telefonica/mistica/compose/README.md:72]: These links were broken, I've fixed them and moved to the table where we have all the components to have only one source of truth

**Review by jdelga — APPROVED** [review]
Great! Thanks!

**Comment by yceballost** [comment]
It's supporting circle + icon?
image

**Comment by jeslat** [comment]
> Is it supporting circle + icon? image

good catch @yceballost, it's supported in the compose implementation but not in the legacy one. I will add it

File: library/src/main/java/com/telefonica/mistica/card/datacard/DataCardView.kt [hunk 4]
```diff
@@ -63,17 +100,55 @@ class DataCardView @JvmOverloads constructor(
...
+            }
+            TYPE_ICON,
+            TYPE_CIRCULAR_ICON,
+            -> {
+                iconImageView.setImageDrawable(icon)
+                iconImageView.visibility = View.VISIBLE
+                assetCircularImageView.visibility = View.GONE
+
+                if (iconType == TYPE_CIRCULAR_ICON) {
+                    DrawableCompat.setTint(DrawableCompat.wrap(icon), context.getThemeColor(R.attr.colorBrand))
```
Comment (jeslat) [inline library/src/main/java/com/telefonica/mistica/card/datacard/DataCardView.kt:117]: One question @yceballost, in the case of the circle + icon, do we have to force the icon color to `colorBrand`? Or is it up to the app to tint this color?

**Review by dpastor — APPROVED** [review]
Thanks a lot @jeslat !! :)

**Review by jdelga — APPROVED** [review]
Good job!

File: library/src/main/java/com/telefonica/mistica/compose/card/datacard/IconPainter.kt [hunk 5]
```diff
@@ -34,24 +36,53 @@ sealed class IconPainter {
...
+                                .wrapContentSize(align = Alignment.Center)
+                        ) {
+                            Image()
+                        }
+                    }
+                    IconType.CIRCULAR_ASSET,
+                    -> {
+                        Circle(
+                            color = backgroundColor
+                        ) {
```
[the remark this answers is not in the document]
Comment (jeslat) [reply library/src/main/java/com/telefonica/mistica/compose/card/datacard/IconPainter.kt:60]: It's forced inside `Circle` component

**Review by dpastor — APPROVED** [review]
good job! :)

**Pull request merged**